### PR TITLE
Api 7.10 Paid media payload and Giveaway updates.

### LINKED ISSUE #4459: [FEATURE] API 7.10
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

API 7.10

> [!IMPORTANT] 
> - Comment here or in the dev group what you want to work on so we can assign you
> - Please copy the relevant part of the checklist from #4459 to your PR and make sure to cover everything :)
> - If this is your first contribution, please be sure to read the [contribution guide](https://docs.python-telegram-bot.org/en/v20.7/contributing.html) and feel free to reach out with any questions via https://t.me/pythontelegrambotchannel/105

### Describe the solution you'd like

- [x] Added updates about purchased paid media, represented by the class [PaidMediaPurchased](https://core.telegram.org/bots/api#paidmediapurchased) and the field purchased_paid_media in the class [Update](https://core.telegram.org/bots/api#update). (Taken by @harshil21)
- [ ] Added the ability to specify a payload in [sendPaidMedia](https://core.telegram.org/bots/api#sendpaidmedia) that is received back by the bot in [TransactionPartnerUser](https://core.telegram.org/bots/api#transactionpartneruser) and purchased_paid_media updates. (Taken by @aelkheir)
- [ ] Added the field prize_star_count to the classes [GiveawayCreated](https://core.telegram.org/bots/api#giveawaycreated), [Giveaway](https://core.telegram.org/bots/api#giveaway), [GiveawayWinners](https://core.telegram.org/bots/api#giveawaywinners) and [ChatBoostSourceGiveaway](https://core.telegram.org/bots/api#chatboostsourcegiveaway).  (Taken by @aelkheir)
- [ ] Added the field is_star_giveaway to the class [GiveawayCompleted](https://core.telegram.org/bots/api#giveawaycompleted).  (Taken by @aelkheir)

### Describe alternatives you've considered

If you want to contribute, you can start from the [Contribution Guide](https://docs.python-telegram-bot.org/en/v20.7/contributing.html).

### Additional context

Please comment below before start working on any part of the update so as to keep things in track and avoid double work.

### PR DESCRIPTION
**Pull request by aelkheir**
Closes parts of #4459
- [x] Added the ability to specify a payload in [sendPaidMedia](https://core.telegram.org/bots/api#sendpaidmedia) that is received back by the bot in [TransactionPartnerUser](https://core.telegram.org/bots/api#transactionpartneruser) and purchased_paid_media updates. (Taken by @aelkheir)
- [x] Added the field prize_star_count to the classes [GiveawayCreated](https://core.telegram.org/bots/api#giveawaycreated), [Giveaway](https://core.telegram.org/bots/api#giveaway), [GiveawayWinners](https://core.telegram.org/bots/api#giveawaywinners) and [ChatBoostSourceGiveaway](https://core.telegram.org/bots/api#chatboostsourcegiveaway).  (Taken by @aelkheir)
- [x] Added the field is_star_giveaway to the class [GiveawayCompleted](https://core.telegram.org/bots/api#giveawaycompleted).  (Taken by @aelkheir)

## Checklist for PRs
- [x] Added ``.. versionadded:: NEXT.VERSION``, ``.. versionchanged:: NEXT.VERSION``, ``.. deprecated:: NEXT.VERSION`` or ``.. versionremoved:: NEXT.VERSION`` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [ ] Documented code changes according to the `CSI standard <https://standards.mousepawmedia.com/en/stable/csi.html>`__
- [ ] Added myself alphabetically to ``AUTHORS.rst`` (optional)
- [ ] Checked the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_ in case of deprecations or changes to documented behavior

**If the PR contains API changes (otherwise, you can ignore this passage)**
- [ ] Checked the Bot API specific sections of the `Stability Policy <https://docs.python-telegram-bot.org/stability_policy.html>`_
- [ ] Created a PR to remove functionality deprecated in the previous Bot API release (`see here <https://docs.python-telegram-bot.org/en/stable/stability_policy.html#case-2>`_)
-  New classes:
   - [x] Added ``self._id_attrs`` and corresponding documentation
   - [x] ``__init__`` accepts ``api_kwargs`` as kw-only
-  Added new shortcuts:
   - [x] In :class:`~telegram.Chat` & :class:`~telegram.User` for all methods that accept ``chat/user_id``
-  If relevant:
   - [x] Link new and existing constants in docstrings instead of hard-coded numbers and strings
   - [x] Added or updated documentation for the changed class(es) and/or method(s)

### CODE REVIEW

File: tests/test_giveaway.py [hunk 1]
```diff
@@ -164,15 +170,41 @@ def test_equality(self, giveaway):
...
+        assert gac.prize_star_count == self.prize_star_count
+
+        assert Giveaway.de_json(None, bot) is None
+
+    def test_to_dict(self, giveaway_created):
+        gac_dict = giveaway_created.to_dict()
+
+        assert isinstance(gac_dict, dict)
+        assert gac_dict["prize_star_count"] == self.prize_star_count
+
```
Comment (Bibo-Joshi) [inline tests/test_giveaway.py:207]: a test_equality test would be needed for new rich equality comparison :)

File: telegram/_chatboost.py [hunk 2]
```diff
@@ -188,14 +188,19 @@ def __init__(self, user: User, *, api_kwargs: Optional[JSONDict] = None):
 class ChatBoostSourceGiveaway(ChatBoostSource):
     """
     The boost was obtained by the creation of a Telegram Premium giveaway. This boosts the chat 4
```
Comment (Bibo-Joshi) [inline telegram/_chatboost.py:190]:
```suggestion
    The boost was obtained by the creation of a Telegram Premium or a Telegram Star giveaway. This boosts the chat 4
```

File: telegram/_giveaway.py [hunk 3]
```diff
@@ -145,13 +156,26 @@ def de_json(
 
 class GiveawayCreated(TelegramObject):
     """This object represents a service message about the creation of a scheduled giveaway.
-    Currently holds no information.
+
```
[the remark this answers is not in the document]
Comment (aelkheir) [reply telegram/_giveaway.py:159]: even though the new param is optional?

**Review by harshil21 — CHANGES_REQUESTED** [review]
just doc fixes

**Review by harshil21 — APPROVED** [review]

**Review by Bibo-Joshi — COMMENTED** [review]
was left hanging in pending …

**Comment by Bibo-Joshi** [comment]
Thank you a lot for the nice and fast PR!

**Pull request merged**